src/crepe/api/UiObject.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "../Component.h"
+
+namespace crepe {
+
+/**
+ * @class UiObject
+ * \brief Represents a UI object in the game, derived from the Component class.
+ */
+class UiObject : public Component {
+public:
+	/**
+     * \brief Constructs a UiObject with the specified game object ID.
+     * \param id The unique ID of the game object associated with this UI object.
+     */
+	UiObject(game_object_id_t id, int width, int height);
+
+	//! The width of the UI object.
+	int width = 0;
+
+	//! The height of the UI object.
+	int height = 0;
+
+public:
+	/**
+     * \brief Retrieves the maximum number of instances allowed for this UI object type.
+     * /return Always returns 1, as only a single instance is allowed.
+     */
+	virtual int get_instances_max() const override { return 1; }
```

Comment:
Hier leg je uit waarom het getal 1 gelijk is aan de hoeveelheid instanties van dit component, niet waarom het überhaupt 1 is. Komt het doordat de knop zelf geen offset heeft en je daardoor anders alleen overlappende knoppen op de zelfde positie kan maken?